test/integration/extension_discovery_integration_test.cc

```diff
@@ -322,6 +382,72 @@ TEST_P(ExtensionDiscoveryIntegrationTest, BasicDefaultMatcher) {
+  test_server_->waitForGaugeGe("listener_manager.workers_started", 1);
+
+  // Update listener and expect it to be warm and use active configuration from the subscription
+  // instead of the default config.
+  listener_config_.set_traffic_direction(envoy::config::core::v3::TrafficDirection::OUTBOUND);
+  sendLdsResponse("updated");
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 2);
+  test_server_->waitForGaugeEq("listener_manager.total_listeners_warming", 0);
+
+  codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
+  Http::TestRequestHeaderMapImpl request_headers{
+      {":method", "GET"}, {":path", "/"}, {":scheme", "http"}, {":authority", "host"}};
+  auto response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  response->waitForEndStream();
+  ASSERT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().getStatusValue());
+}
+
+TEST_P(ExtensionDiscoveryIntegrationTest, ReuseExtensionConfigInvalid) {
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  addDynamicFilter("foo", false);
+  initialize();
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 1);
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initializing);
+  registerTestServerPorts({"http"});
+  sendXdsResponseWithFullYaml("foo", "1", denyPrivateConfigWithMatcher());
+  test_server_->waitForCounterGe("http.config_test.extension_config_discovery.foo.config_reload",
+                                 1);
+  test_server_->waitUntilListenersReady();
+  test_server_->waitForGaugeGe("listener_manager.workers_started", 1);
+
+  // Remove matcher filter type URL, invalidating the subscription last config.
+  auto hcm_config = MessageUtil::anyConvert<
+      envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager>(
+      listener_config_.filter_chains(0).filters(0).typed_config());
+  hcm_config.mutable_http_filters(0)->mutable_config_discovery()->mutable_type_urls()->Clear();
+  hcm_config.mutable_http_filters(0)->mutable_config_discovery()->add_type_urls(
+      "type.googleapis.com/test.integration.filters.SetResponseCodeFilterConfig");
+  listener_config_.mutable_filter_chains(0)->mutable_filters(0)->mutable_typed_config()->PackFrom(
+      hcm_config);
+
+  sendLdsResponse("updated");
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 2);
+  test_server_->waitForGaugeEq("listener_manager.total_listeners_warming", 0);
+
+  // Should be using the default config (403)
+  codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
+  Http::TestRequestHeaderMapImpl request_headers{
+      {":method", "GET"}, {":path", "/"}, {":scheme", "http"}, {":authority", "host"}};
+  auto response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  response->waitForEndStream();
+  ASSERT_TRUE(response->complete());
+  EXPECT_EQ("403", response->headers().getStatusValue());
+}
```

Comment:
Do we expect some metric to be incremented in this case to cover the failed config application?